### PR TITLE
#6759 showstopper one-to-one inverse not being loaded with correct identifier restrictions when defining `joinColumn`

### PR DESCRIPTION
**Pull request by Ocramius**
Fixes #6759

### CODE REVIEW

File: lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php [hunk 1]
```diff
@@ -802,13 +804,11 @@ public function loadOneToOneEntity(array $assoc, $sourceEntity, array $identifie
 
             // unset the old value and set the new sql aliased value here. By definition
             // unset($identifier[$targetKeyColumn] works here with how UnitOfWork::createEntity() calls this method.
```
Comment (alcaeus) [inline lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php]: Comment is out of date - there's no `unset` left.

File: tests/Doctrine/Tests/ORM/Functional/OneToOneInverseSideLoadAfterDqlQueryTest.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                $this->_em->getClassMetadata(OwningSide::class),
+                $this->_em->getClassMetadata(InverseSide::class),
+            ]);
+        } catch(ToolsException $e) {
+            // ignored
+        }
+    }
+
+    /**
+     * @group #6759
```
Comment (lcobucci) [inline tests/Doctrine/Tests/ORM/Functional/OneToOneInverseSideLoadAfterDqlQueryTest.php]: All the other test cases don't have the `#`, I'd remove it for consistency.

**Review by lcobucci — CHANGES_REQUESTED** [review]
Just this small thing

**Review by lcobucci — APPROVED** [review]

**Comment by lcobucci** [comment]
@Ocramius 🚢

**Pull request merged**